client/config/toml.go

```diff
@@ -27,41 +26,42 @@ broadcast-mode = "{{ .ClientConfig.BroadcastMode }}"
 trace = "{{ .ClientConfig.Trace }}"
 `
 
-var configTemplate *template.Template
-
-func InitConfigTemplate() {
-	var err error
-
+// InitConfigTemplate initiates config template that will be used in
+// WriteConfigFile
+func InitConfigTemplate() *template.Template {
 	tmpl := template.New("clientConfigFileTemplate")
-
-	if configTemplate, err = tmpl.Parse(defaultConfigTemplate); err != nil {
+	configTemplate, err := tmpl.Parse(defaultConfigTemplate)
+	if err != nil {
 		panic(err)
 	}
 
+	return configTemplate
 }
 
 // ParseConfig retrieves the default environment configuration for the
 // application.
-func ParseConfig() *ClientConfig {
+func ParseConfig(v *viper.Viper) (*ClientConfig, error) {
 	conf := DefaultClientConfig()
-	 _ = viper.Unmarshal(conf)
+	if err := v.Unmarshal(conf); err != nil {
+		return nil, err
+	}
 
-	return conf
+	return conf, nil
 }
 
 // WriteConfigFile renders config using the template and writes it to
 // configFilePath.
-func WriteConfigFile(configFilePath string, config *ClientConfig) {
+func WriteConfigFile(cfgFile string, config *ClientConfig, configTemplate *template.Template) {
```

Comment:
I have added third parameter in WriteConfigFile to write template into file, since I got rid of configTemplate global variable above.